src/GafferImage/Catalogue.cpp

```diff
@@ -1148,57 +1214,100 @@ void Catalogue::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outp
 void Catalogue::hash( const Gaffer::ValuePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
 {
 	ImageNode::hash( output, context, h );
-	if( output == internalImageIndexPlug() )
+	if( output == invalidImageTextPlug() )
+	{
+		h.append( context->get<std::string>( g_imageNameContextName, g_emptyString ) );
+	}
+	else if( output == internalImageIndexPlug() )
 	{
-		const std::string imageName = context->get<std::string>( "catalogue:imageName", "" );
+		const std::string &imageName = context->get<std::string>( g_imageNameContextName, g_emptyString );
 		if( imageName.empty() )
 		{
 			imageIndexPlug()->hash( h );
 		}
 		else
 		{
-			h.append( imageName );
-			for( const auto &image : Image::Range( *imagesPlug() ) )
+			if( boost::starts_with( imageName, g_outputPrefix ) )
 			{
-				image->namePlug()->hash( h );
+				h.append( imageName.substr( g_outputPrefix.size() ) );
+				for( const auto &image : Image::Range( *imagesPlug() ) )
+				{
+					image->outputIndexPlug()->hash( h );
+				}
+			}
+			else
+			{
+				h.append( imageName );
+				for( const auto &image : Image::Range( *imagesPlug() ) )
+				{
+					image->namePlug()->hash( h );
+				}
 			}
 		}
 	}
 }
 
 void Catalogue::compute( ValuePlug *output, const Context *context ) const
 {
+	if( output == invalidImageTextPlug() )
+	{
+		const std::string &imageName = context->get<std::string>( g_imageNameContextName, g_emptyString );
+		int outputIndex = 0;
+
+		if( boost::starts_with( imageName, g_outputPrefix ) )
+		{
+			outputIndex = std::stoi( imageName.substr( g_outputPrefix.size() ) );
+		}
+
+		std::string result;
+		if( outputIndex > 0 )
+		{
+			result = "Catalogue Output " + std::to_string( outputIndex );
+		}
+		else
+		{
+			result = "Catalogue Unknown Image: \"" + imageName + "\"";
```

Comment:
Maybe `Catalogue : Unknown Image "<imageName>"`? "Catalogue Unknown Image" without the separator doesn't parse very well in my brain...